### DISCUSSION TITLE
how to make shortcuts work as in bash/readline?

### DISCUSSION THREAD

**Discussion by tkossak** [leo-arch/clifm#196] [Q&A] [answered]
Hi,

Is it possible to make `ctrl-d` or `alt-f` behave like in bash (readline) or other shells?

Current behavior in clifm:
When I press `ctrl-d`, it exits clifm, no matter if i'm in the middle of writing new command or nothing is entered yet.
When I press `alt-f` it moves one character forward (instead of one word).

What I would like it to behave:
`ctrl-d` should send EOF (exit shell) only when nothing is entered in the command line.
When I enter any text, `ctrl-d` should work like `delete` button.
`alt-f` should move cursor to next word (not next character).
That's how it works in bash. Is it possible for these shortcuts to work the same in clifm?

Thank you.

I use clifm v1.10.

**Comment by leo-arch** [comment]
Hi @tkossak. Thanks for reporting. I'll take a look at it as soon as I can (seems a reasonable demand btw).

**Comment by leo-arch** [comment, marked as answer]
Well, It was easier to implement than I thought it would. Please give it a try a let me know if it works fine for you too.

EDIT: Just in case, you need to [recompile/install the latest git version](https://github.com/leo-arch/clifm#installing-from-source-linuxbsd).

**Comment by tkossak** [comment]
Wow, I didn't expect such fast response, thank you very much (also for this wonderful software - I'm just learning everything but I already like it very much).

As for the keys - `alt-f` works perfectly now, but `ctrl-d` works as backspace instead of delete :) (`ctrl-h` is for backspace in readline)

**Reply by leo-arch** [reply]
> ctrl-d works as backspace instead of delete

Ok. I'll fix it.

**Reply by leo-arch** [reply]
Done.

**Reply by tkossak** [reply]
it works, thank you very much :)
